### DISCUSSION TITLE
Consider adding `IEqualityOperators<TSelf, TOther, TResult>` and `IClonable` on `record` types

### DISCUSSION THREAD

**Discussion by ero-qt** [dotnet/csharplang#7221] [Language Ideas]
Today, `record` types only implement the `IEquatable<T>` interface by default:
```cs
var interfaces = typeof(R).GetInterfaces().Select(t => t.Name);
Console.WriteLine(string.Join('\n', interfaces));

record R;
```
```
IEquatable`1
```

However, all `record` types also implement `operator==`, `operator!=`, and `Clone`:
```cs
var methods = typeof(R).GetMethods().Select(mi => mi);
Console.WriteLine(string.Join('\n', methods));

record R;
```
```
System.String ToString()
Boolean op_Inequality(R, R)
Boolean op_Equality(R, R)
Int32 GetHashCode()
Boolean Equals(System.Object)
Boolean Equals(R)
R <Clone>$()
System.Type GetType()
```

These methods could be covered by implementing the aforementioned `IEqualityOperators<TSelf, TOther, TResult>` and `IClonable` interfaces respectively while also giving users a bit wider usage range.

**Comment by Youssef1313** [comment]
I don't think `IClonable` should be implemented by records. It's a discouraged interface anyway.

**Reply by colejohnson66** [reply]
It also returns an `object`, not a `T` like records do.

**Comment by CyrusNajmabadi** [comment]
This is a language request.  Moving to csharplang.

**Comment by RenderMichael** [comment]
It would make sense to also implement `IStructuralEquatable`, like Tuples, ValueTuples, and arrays do. It would fit the "grow-up story" ValueTuples => records
